java/bulk-export/bulk-export-core/src/main/java/sleeper/bulkexport/core/recordretrieval/BulkExportQuerySplitter.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright 2022-2024 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.bulkexport.core.recordretrieval;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import sleeper.bulkexport.core.model.BulkExportLeafPartitionQuery;
+import sleeper.bulkexport.core.model.BulkExportQuery;
+import sleeper.core.partition.Partition;
+import sleeper.core.partition.PartitionTree;
+import sleeper.core.properties.table.TableProperties;
+import sleeper.core.range.Region;
+import sleeper.core.statestore.StateStore;
+import sleeper.core.statestore.StateStoreException;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static sleeper.core.properties.table.TableProperty.QUERY_PROCESSOR_CACHE_TIMEOUT;
+import static sleeper.core.properties.table.TableProperty.TABLE_ID;
+
+/**
+ * Runs bulk export queries against a Sleeper table by querying the state store,
+ * data files directly and splits up a BulkExportQuery into multiple
+ * BulkExportLeafPartitionQuery.
+ */
```

Comment:
It doesn't look like it queries the data files directly, it looks like that part needs removing?

This is a bit long for a summary sentence. Can we either shorten what it does and avoid referencing the class names, or split the sentence into two?